internal/controller/bmc_controller.go

```diff
@@ -162,6 +162,7 @@ func (r *BMCReconciler) discoverServers(ctx context.Context, log logr.Logger, bm
 	for i, s := range servers {
 		server := &metalv1alpha1.Server{}
 		server.Name = bmcutils.GetServerNameFromBMCandIndex(i, bmcObj)
+		server.Labels = bmcObj.Labels
```

Comment:
Also shouldn't we append the labels in the mutate function e.g. if the `Server` has already labels?